decide/lic4.cpp

```diff
@@ -0,0 +1,105 @@
+#include <decide/decide.hpp>
+
+/*There exists at least one set of Q_PTS (consecutive data points) that lie in more than QUADS
+(quadrants). Where there is ambiguity as to which quadrant contains a given point, priority
+of decision will be by quadrant number, i.e., I, II, III, IV. For example, the data point (0,0)
+is in quadrant I, the point (-l,0) is in quadrant II, the point (0,-l) is in quadrant III, the point
+(0,1) is in quadrant I and the point (1,0) is in quadrant I.*/
+
+int quad(Coordinate point){
+    if (point.x>=0){
+        if (point.y>=0){
+            return 1;
+        }
+        else{
+            return 4;
+        }
+    }
+    else{
+        if (point.y>=0){
+            return 2;
+        }
+        else{
+            return 3;
+        }
+    }
+}
+
+bool unique(int arr[], int goal){
+    int first=0;
+    int second=0;
+    int third=0;
+    int forth=0;
+    for (int i=0; i < sizeof(arr);i++){
+        switch (arr[i])
+        {
+        case 1:
+            if (first==0){
+                 first++;
+            }
+            break;
+        case 2:
+            if (second==0){
+                 second++;
+            }
+            break;
+        case 3:
+            if (third==0){
+                 third++;
+            }
+            break;
+        case 4:
+            if (forth==0){
+                 forth++;
+            }
+            break;
+        }
+    }
+    if ((first+second+third+forth)>=goal){
```

Comment:
Since `goal` is `param.QUADS`, this needs to use `>`, not `>=`, as per the spec ("more than").